core/src/transport/memory.rs

```diff
@@ -311,5 +313,43 @@ mod tests {
         assert!(transport.dial("/memory/810172461024613".parse().unwrap()).is_ok());
     }
 
-    // TODO: test that is actually works
+    #[test]
+    fn communicating_between_dialer_and_listener() {
+        let msg = [1, 2, 3];
+
+        // Setup listener.
+
+        let rand_port = rand::random::<u64>() + 1;
```

Comment:
Why this `+ 1` by the way?

Reply:
Well, ... `0` would instruct the OS to bind to any port. I guess it is very unlikely. Do you want me to remove it @tomaka?

Reply:
Well, it is equally likely that we randomly generate `u64::max_value()` and that the `+1` either panics or wraps to 0.

Reply:
I would go with `saturating_add(1)`.

Reply:
Here's the "Apply suggestion" button ;)

```suggestion
        let rand_port = rand::random::<u64>().saturating_add(1);
```

Reply:
Thanks @romanb and @tomaka!